core/src/main/java/org/jruby/prism/parser/ParserPrism.java

```diff
@@ -176,20 +184,23 @@ private byte[] loadFully(String fileName, InputStream in) {
     private byte[] parse(byte[] source, int sourceLength, byte[] metadata) {
         long time = 0;
         if (ParserManager.PARSER_TIMING) time = System.nanoTime();
-        ParserBindingPrism.Buffer buffer = new ParserBindingPrism.Buffer(jnr.ffi.Runtime.getRuntime(prismLibrary));
-        prismLibrary.pm_buffer_init(buffer);
-        prismLibrary.pm_serialize_parse(buffer, source, sourceLength, metadata);
+        assert(source.length == sourceLength); // TODO: throw a better exception
+
+        byte[] options = ParsingOptions.serialize(new byte[0], 0, metadata, true, false, (byte) 0, new byte[0][0][0]);
```

Comment:
quick comment @enebo , I tested by using `metadata = "UTF-8".getBytes();` and it looks better (it doesn't throw exceptions) but a lot of `gemspec`s are failing to parse.

Reply:
@andreaTP yeah I think gemspecs won't parse because `new byte[0][0][0])` needs to be a list of variable scopes and not an empty value.  If the gemspec sees a word while parsing which represents a variable from a captured scope then it will make a variable get otherwise it ends up being a argument-less call.

Don't be too concerned here.  I will change this to using my version of this serialize method.  The one shipped here is used by truffleRuby.  I don't want to create a byte[][][] so I do it a little differently to avoid that extra allocation.

Reply:
Ok, thanks for the explanation, makes sense to me, feel free to close this PR and just use it as an inspiration when time comes. The compatibility should be guaranteed by the CI in the official Prism repo 👍

Reply:
@andreaTP I just updated it in place which is new for me but it was helpful.  I think I correct metadata being properly setup but the gemspecs still fail with invalid token being reported on the final line of the file being eval'd. I will land this and communicate elsewhere.